src/delta_lake/delta_service.py

```diff
@@ -33,6 +35,57 @@
     "vacuum",
 }
 
+DISALLOW_SQL_META_CHARS = {
+    "--",
+    "/*",
+    "*/",
+    ";",
+    "\\",
+}
+
+ALLOWED_STATEMENTS = {
+    "select",
+}
+
+FORBIDDEN_POSTGRESQL_SCHEMAS = {
+    "pg_",
+    "pg_catalog",
+    "information_schema",
+}
+
+
+def _check_query_is_valid(query: str) -> bool:
+    """
+    Check if a query is valid.
+    """
+
+    statements = sqlparse.parse(query)
```

Comment:
I assume if the sql is invalid this throws an error? Do you need to catch and rethrow with a SparkQueryError?

Reply:
👍

Reply:
Actually, looking at the docs, this parser doesn't validate the sql structure so I assume it won't throw an error. That seems bad: https://github.com/andialbrecht/sqlparse/blob/master/README.rst

This seems better: https://github.com/tobymao/sqlglot

But trying to figure out how to get it to do something simple like find all the sql statements in the parse tree seems like a PITA

Reply:
Added a note, good enough for now